### PR TITLE
Compute the Kuznyechik tables instead of hardcoding them

### PR DESCRIPTION
**Pull request by randombit**

### CODE REVIEW

File: src/lib/block/kuznyechik/kuznyechik.cpp [hunk 1]
```diff
@@ -47,4103 +48,91 @@ alignas(256) const uint8_t IS[256] = {
...
-                                  {0xfcd3141c7ba10ef0, 0x62bc896ce66409c0}, {0x6526890d6033e7a3, 0xdecfe2390bbd753d},
-                                  {0x261b71fa9e277d09, 0xe9f4c5942a1fa2b5}, {0x4817cf086776040d, 0x2cbf2b2883ef06da},
-                                  {0xb4c4db141cd70afd, 0x4e03a244658b0f1a}, {0x6a2d93cd6169871b, 0xf8df8c7f7e21251c},
-                                  {0x556996ca0fd06527, 0x01f03f64a638b62c}, {0x16546e3df1c4ff8d, 0x36cb18c9879a61a4},
-                                  {0xcbd54b65e668d3f2, 0xe178fe32531c5b16}, {0x1275430269fc0192, 0x0b5f7a0a504be0d7},
-                                  {0x91b7c76fe8e2d66d, 0xc698af1080b8bd1b}, {0x1ce7c3bd4ea8bf5d, 0xb36aed0c6072019a},
-                                  {0x06d0dac1d42481f1, 0xc2de53435d5820ab}, {0x0c6377416b48c121, 0x477fa686bab04095},
-                                  {0x2e592b846d574237, 0x931f01d1477e6353}, {0x2fc0b1fb4b599c40, 0x0d3af89102dbd23f},
-                                  {0xdb51ff99c388ad8e, 0x156db5b889de1a19}, {0x5bfb16752884dbe8, 0xb9c5a86296015761},
-                                  {0xe9ef17a0e07750e4, 0x35185965aed278d0},
```
Comment (reneme) [inline src/lib/block/kuznyechik/kuznyechik.cpp:135]:
```suggestion
constexpr auto T = T_table();
constexpr auto IT = IT_table();
```

?

Comment (randombit) [reply src/lib/block/kuznyechik/kuznyechik.cpp]: That was the plan but it hits constexpr expression limits with Clang.

However I just pushed a new versoin that at least for my Clang does allow compile time initialization. Let's see what CI thinks..

Comment (randombit) [reply src/lib/block/kuznyechik/kuznyechik.cpp]: It worked!

**Comment by mouse07410** [comment]
Naive question: what was wrong with  precomputed tables?

**Comment by randombit** [comment]
Nothing "wrong" with them, they just take up a lot of (hard to compress) bytes in a source code tarball.

**Review by reneme — APPROVED** [review]
No thorough review against the spec, code itself looks fine.

**Pull request merged**